backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/JWTgenerator.java

```diff
@@ -0,0 +1,135 @@
+package org.molgenis.emx2;
+
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.emx2.ColumnType.STRING;
+import static org.molgenis.emx2.sql.SqlDatabase.ADMIN_USER;
+
+import com.nimbusds.jose.*;
+import com.nimbusds.jose.crypto.MACSigner;
+import com.nimbusds.jose.crypto.MACVerifier;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import java.security.SecureRandom;
+import java.util.Calendar;
+import java.util.Date;
+import org.molgenis.emx2.utils.EnvironmentProperty;
+
+public class JWTgenerator {
+  private static byte[] sharedSecret;
+  private static MACSigner signer;
+  private static JWSVerifier verifier;
+
+  private static void init() {
+    try {
+      sharedSecret = null;
+      String key =
+          (String)
+              EnvironmentProperty.getParameter(Constants.MOLGENIS_JWT_SHARED_SECRET, null, STRING);
+      if (key != null) {
+        // try to parse from environment variable
+        sharedSecret = key.getBytes();
+      } else {
+        // Generate random 256-bit (32-byte) shared secret
+        sharedSecret = new byte[32];
+        new SecureRandom().nextBytes(sharedSecret);
+      }
+
+      // check enough bytes
+      if (sharedSecret.length < 32) {
+        throw new MolgenisException(
+            Constants.MOLGENIS_JWT_SHARED_SECRET
+                + " was not secure enough, should be 32 bytes/256bit");
+      }
+
+      // Create HMAC signer
+      signer = new MACSigner(sharedSecret);
+      verifier = new MACVerifier(sharedSecret);
+    } catch (Exception kle) {
+      throw new MolgenisException(kle.getMessage());
+    }
+  }
+
+  /*Token should either have id */
+  public static String createNamedTokenForUser(Database db, String user, String tokenId) {
+    Calendar c = Calendar.getInstance();
+    c.setTime(new Date());
+    c.add(Calendar.YEAR, 10);
```

Comment:
why 10, needs code comment and maybe setting , maybe inject date to make component less statefull

Reply:
I actually meant it to be endless. Comment added for that.